src/base.c

```diff
@@ -98,7 +330,14 @@ static const
 R_CallMethodDef callMethods[] = {
   {"kx_r_open_connection", (DL_FUNC) &kx_r_open_connection, -1},
   {"kx_r_close_connection", (DL_FUNC) &kx_r_close_connection, -1},
-  {"kx_r_execute", (DL_FUNC) &kx_r_execute, -1},
+  {"kx_r_execute0", (DL_FUNC) &kx_r_execute0, -1},
```

Comment:
I do not have clang-format. What is that for?

>>> Maybe worth trying approach similar to what @jameslittle did here with ... args
Repetitive boilerplate code in C++ is indeed a problem. Usually you can resolve it by using preprocessor template technique like this library: http://www.boost.org/doc/libs/1_64_0/libs/preprocessor/doc/index.html . But this will introduce another dependency on rkdb. I think I will leave the implementation for now.